samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -50,25 +50,30 @@
   private final String hostName;
 
   private AtomicBoolean isLeader = new AtomicBoolean(false);
-  private final IZkDataListener zkLeaderElectionListener;
+  private final IZkDataListener zkLeaderElectionInternalListener;
+  ZkLeaderElectorListener zkLeaderElectorExtListener;
   private String currentSubscription = null;
   private final Random random = new Random();
 
   @VisibleForTesting
-  ZkLeaderElector(String processorIdStr, ZkUtils zkUtils, IZkDataListener leaderElectionListener) {
+  ZkLeaderElector(String processorIdStr,
+      ZkUtils zkUtils,
+      ZkLeaderElectorListener zkLeaderElectorExtListener,
```

Comment:
I am confused with these 2 interfaces. I understand the reason you added another listener is to serve as an interface between ZkController and ZkLeaderElector module because we do "tryBecomeLeader" asynchronously. Is there a better way of doing this?
If I were to draw a block diagram explaining the interactions between the different components (like ZkControllerImpl, ZkControllerListener, ZkLeaderElectorListener etc), it seems complicated. I think we should simplify this interaction.